### PR TITLE
 mrbayes: modernize 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/biology/mrbayes/default.nix [hunk 1]
```diff
@@ -22,8 +38,9 @@ stdenv.mkDerivation rec {
       MrBayes uses a simulation technique called Markov chain Monte Carlo (or
       MCMC) to approximate the posterior probabilities of trees.
     '';
-    license     = licenses.gpl2;
-    homepage    = "http://mrbayes.csit.fsu.edu/";
+    maintainers = with maintainers; [ ];
+    license = licenses.gpl2;
+    homepage = "http://mrbayes.csit.fsu.edu/";
```
Comment (prusnak) [inline pkgs/applications/science/biology/mrbayes/default.nix:43]:
```suggestion
    homepage = "https://nbisweden.github.io/MrBayes/";
```

File: pkgs/applications/science/biology/mrbayes/default.nix [hunk 2]
```diff
@@ -1,18 +1,34 @@
-{lib, stdenv, fetchurl, readline}:
+{ lib, stdenv, fetchurl, readline }:
+
+# FIXME: replace Makefile so we can build MPI & MAC support
 
 stdenv.mkDerivation rec {
-  # FIXME: replace Makefile so we can build MPI & MAC support
+  pname = "mrbayes";
+  version = "3.1.2";
```
Comment (prusnak) [inline pkgs/applications/science/biology/mrbayes/default.nix:7]: Let's update to `3.2.7a` which is the latest release.

File: pkgs/applications/science/biology/mrbayes/default.nix [hunk 3]
```diff
@@ -1,18 +1,34 @@
...
 
-  name = "mrbayes-3.1.2";
   src = fetchurl {
```
Comment (prusnak) [inline pkgs/applications/science/biology/mrbayes/default.nix:9]: Let's fetch from Github: https://github.com/NBISweden/MrBayes

File: pkgs/applications/science/biology/mrbayes/default.nix [hunk 4]
```diff
@@ -22,8 +38,9 @@ stdenv.mkDerivation rec {
       MrBayes uses a simulation technique called Markov chain Monte Carlo (or
       MCMC) to approximate the posterior probabilities of trees.
     '';
-    license     = licenses.gpl2;
-    homepage    = "http://mrbayes.csit.fsu.edu/";
+    maintainers = with maintainers; [ ];
+    license = licenses.gpl2;
```
Comment (prusnak) [inline pkgs/applications/science/biology/mrbayes/default.nix:42]:
```suggestion
    license = licenses.gpl2Plus;
```

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 131531` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mrbayes</li>
  </ul>
</details>

**Comment by siraben** [comment]
I agree with @prusnak's suggestions.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131531` at 66719382 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>mrbayes</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/science/biology/mrbayes/default.nix:24:3:
```
   |
24 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/science/biology/mrbayes/default.nix:42:5:
```
   |
42 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/science/biology/mrbayes/default.nix:41:5:
```
   |
41 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/applications/science/biology/mrbayes/default.nix [hunk 5]
```diff
@@ -1,18 +1,18 @@
-{lib, stdenv, fetchurl, readline}:
+{ lib, stdenv, fetchFromGitHub, readline }:
 
 stdenv.mkDerivation rec {
-  # FIXME: replace Makefile so we can build MPI & MAC support
+  pname = "mrbayes";
+  version = "3.2.7";
```
Comment (prusnak) [inline pkgs/applications/science/biology/mrbayes/default.nix:5]: According to https://nbisweden.github.io/MrBayes/download.html the latest is 3.2.7a

```suggestion
  version = "3.2.7a";
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/biology/mrbayes/default.nix (hunk 5 above)
Comment (prusnak) [reply pkgs/applications/science/biology/mrbayes/default.nix:5]: Why has this been resolved and merged without addressing the issue?

Comment (SuperSandro2000) [reply pkgs/applications/science/biology/mrbayes/default.nix:5]: there is not such release on github. assumed this was a typo.

Comment (SuperSandro2000) [reply pkgs/applications/science/biology/mrbayes/default.nix:5]: basic sorting is hard...

image

**Pull request merged**